From: Russell King - ARM Linux <linux@arm.linux.org.uk>
Subject: Fwd: patch for eepro100
To: arjanv@redhat.com
Cc: quintela@mandrakesoft.com, jgarzik@mandrakesoft.com
Date: Mon, 25 Feb 2002 19:35:11 +0000

You people may be very interested in this!

Subject: patch for eepro100 
To: "'saw@saw.sw.com.sg'" <saw@saw.sw.com.sg>,
        "'Russell King - ARM Linux'" <linux@arm.linux.org.uk>
Date: Mon, 25 Feb 2002 11:24:48 -0800


Hello,
I found a problem with the eepro100 - seems to be in all current versions of
the driver.  I duplicated this issue on an SA110 platform with 16MB of RAM,
82559ER, but the bug is architecture independent.  To duplicate, from a PC
attached via 100Mbit switch, full duplex, type:
# ping -f <ip_address_of_82559> -s 45000 -l 50000

You will see the eepro100 driver lose all it's resources and hit the code
block with puts out the "too much work at interrupt" debug message.  At this
point the card is left with receives disabled and loses all receive
connectivity.  This patch addresses the bug.
Kristen Carlson



diff -urN -X /usr/src/dontdiff linux.orig/drivers/net/eepro100.c
linux/drivers/net/eepro100.c
--- linux.orig/drivers/net/eepro100.c	Fri Jan 18 16:05:52 2002
+++ linux/drivers/net/eepro100.c	Fri Feb 22 10:41:35 2002
@@ -1732,7 +1732,7 @@
 			/* Clear all interrupt sources. */
 			/* Will change from 0xfc00 to 0xff00 when we start handling
 			   FCP and ER interrupts --Dragan */
-			outw(0xfc00, ioaddr + SCBStatus);
+			outw(status & 0xfc00, ioaddr + SCBStatus);
 			break;
 		}
 	} while (1);


